dissect/target/plugins/os/windows/registry.py

```diff
@@ -309,3 +324,14 @@ def get_user(self, key: RegistryKey) -> WindowsUserRecord:
         details = self._hives_to_users.get(key.hive)
         if details:
             return details.user
+
+    @internal
+    def glob_ext(self, pattern: str) -> Iterator[KeyCollection]:
```

Comment:
Would it not be nicer if this function returns the combined key collection? That way it would be more compatible with the existing API and it would be somewhat easier to add globbing to all registry functions?

Reply:
I think the other functions should preferably return an iterator instead of the lists they do now. Performance wise, `glob_ext()` imo should return an iterator.